webrtc-c/canary/src/CloudwatchMonitoring.cpp

```diff
@@ -96,4 +95,82 @@ VOID CloudwatchMonitoring::pushICEHolePunchingDelay(UINT64 delay, StandardUnit u
+
+    averageFramesRateDatum.SetMetricName("FramesPerSecond");
+    averageFramesRateDatum.SetValue(pOutboundRtpStats->averageFramesSentPerSecond);
+    averageFramesRateDatum.SetUnit(StandardUnit::Count_Second);
+    averageFramesRateDatum.AddDimensions(this->channelDimension);
+    this->push(averageFramesRateDatum);
+
+    nackRateDatum.SetMetricName("NackRate");
+    nackRateDatum.SetValue(pOutboundRtpStats->nacksPerSecond);
+    nackRateDatum.SetUnit(StandardUnit::Count_Second);
+    nackRateDatum.AddDimensions(this->channelDimension);
+    this->push(nackRateDatum);
+
+    retransmissionPercentDatum.SetMetricName("PercentageFramesRetransmitted");
+    retransmissionPercentDatum.SetValue(pOutboundRtpStats->retxBytesPercentage);
+    retransmissionPercentDatum.SetUnit(StandardUnit::Percent);
+    retransmissionPercentDatum.AddDimensions(this->channelDimension);
+    this->push(retransmissionPercentDatum);
+}
+
+VOID CloudwatchMonitoring::pushInboundRtpStats(Canary::PIncomingRTPMetricsContext pIncomingRtpStats)
+{
+    MetricDatum incomingBitrateDatum, incomingPacketRate, incomingFrameDropRateDatum;
+
+    incomingBitrateDatum.SetMetricName("IncomingBitRate");
+    incomingBitrateDatum.SetValue(pIncomingRtpStats->incomingBitRate);
+    incomingBitrateDatum.SetUnit(StandardUnit::Kilobits_Second);
+    incomingBitrateDatum.AddDimensions(this->channelDimension);
+    this->push(incomingBitrateDatum);
+
+    incomingPacketRate.SetMetricName("IncomingPacketRate");
+    incomingPacketRate.SetValue(pIncomingRtpStats->packetReceiveRate);
+    incomingPacketRate.SetUnit(StandardUnit::Count_Second);
+    incomingPacketRate.AddDimensions(this->channelDimension);
+    this->push(incomingPacketRate);
+
+    incomingFrameDropRateDatum.SetMetricName("IncomingFrameDropRate");
+    incomingFrameDropRateDatum.SetValue(pIncomingRtpStats->framesDroppedPerSecond);
+    incomingFrameDropRateDatum.SetUnit(StandardUnit::Count_Second);
+    incomingFrameDropRateDatum.AddDimensions(this->channelDimension);
+    this->push(incomingFrameDropRateDatum);
+}
+
+VOID CloudwatchMonitoring::pushEndToEndMetrics(Canary::PEndToEndMetricsContext pEndToEndMetricsContext)
+{
+    MetricDatum endToEndLatencyDatum, dataMatchDatum;
+    endToEndLatencyDatum.SetMetricName("EndToEndFrameLatency");
+    endToEndLatencyDatum.SetStorageResolution(1);
+    endToEndLatencyDatum.SetUnit(StandardUnit::Milliseconds);
+    endToEndLatencyDatum.AddDimensions(this->channelDimension);
+    // This debug will be removed in the future
+    for (auto it = pEndToEndMetricsContext->frameLatency.begin(); it != pEndToEndMetricsContext->frameLatency.end(); it++) {
+        DLOGD("Latency: %lf", *it);
+    }
+    DLOGD("Vector size: %lu", pEndToEndMetricsContext->frameLatency.size());
+    endToEndLatencyDatum.SetValues(pEndToEndMetricsContext->frameLatency);
+    this->push(endToEndLatencyDatum);
+
+    dataMatchDatum.SetMetricName("FrameDataMatch");
+    dataMatchDatum.SetStorageResolution(1);
```

Comment:
Why do we need to change the resolution to 1 second?

Reply:
I had added it earlier because I was testing emitting metrics at a 1 second interval and I wanted to set high resolution. Don't need it anymore. Removed.